crates/libcgroups/src/common.rs

```diff
@@ -216,6 +216,21 @@ fn create_v2_cgroup_manager(_cgroup_path: PathBuf) -> Result<Box<dyn CgroupManag
     bail!("cgroup v2 feature is required, but was not enabled during compile time");
 }
 
+pub fn rootless_required() -> bool {
+    if !nix::unistd::geteuid().is_root() {
+        return true;
+    }
+
+    let uid_map_path = "/proc/self/uid_map";
+    let content = fs::read_to_string(uid_map_path)
+        .unwrap_or_else(|_| panic!("failed to read {}", uid_map_path));
+    if !content.contains("4294967295") {
```

Comment:
Maybe we should add a comment explaining the meaning of this ... like in https://github.com/opencontainers/runc/blob/v1.1.3/libcontainer/userns/userns_linux.go#L30